### PR TITLE
feat: add quoting support

### PR DESCRIPTION
**Pull request by cstruct**
Add quoting support both where Presto and Hive style is used.

From request by @nicor88

### CODE REVIEW

**Comment by nicor88** [comment]
@cstruct could you fix the linting issues?

**Comment by cstruct** [comment]
Yes, of course! I'll fix it tomorrow when I'm home again.

**Comment by cstruct** [comment]
Sorry for the wait! Forgot to do it during the weekend :)

File: dbt/adapters/athena/relation.py [hunk 1]
```diff
@@ -13,9 +13,24 @@ class AthenaIncludePolicy(Policy):
 
 @dataclass(frozen=True, eq=False, repr=False)
 class AthenaRelation(BaseRelation):
-    quote_character: str = ""
+    quote_character: str = '"'  # Presto quote character
     include_policy: Policy = AthenaIncludePolicy()
 
+    def render_hive(self):
```
[the remark this answers is not in the document]
Comment (cstruct) [reply dbt/adapters/athena/relation.py:19]: To be honest I don't really see what a unit test gives us here. Maybe a functional test make sense if I understand their purpose. Either way the test setup fails for me locally and I don't feel like I have the time to look into it at the moment. `Allow edits and access to secrets by maintainers` is checked for this PR so if a maintainer wants to do it I don't have an issue with that.

[the remark this answers is not in the document]
Comment (cstruct) [reply dbt/adapters/athena/relation.py:19]: I added two unit tests even though I don't really see the value. Functional tests I think would be great but I don't have the time to look into that at the moment.

[the remark this answers is not in the document]
Comment (nicor88) [reply dbt/adapters/athena/relation.py:19]: It's ok, we might cover them later.

**Comment by cstruct** [comment]
@nicor88 I'm not familiar with the iceberg implementation but I pushed a commit with a guess for a fix. Could you try again? If this doesn't work I'll have to create a test setup and do som proper debugging.

**Comment by nicor88** [comment]
@cstruct we can help you out, most of the code is in [helpers_iceberg](https://github.com/dbt-athena/dbt-athena/blob/main/dbt/include/athena/macros/materializations/models/helpers_iceberg.sql) and in  [incremental](https://github.com/dbt-athena/dbt-athena/blob/main/dbt/include/athena/macros/materializations/models/incremental/incremental.sql) (not sure if it's needed here, it might be needed too)

**Comment by cstruct** [comment]
@nicor88 alright it doesn't seem to involved. Unfortunately I've managed to bork my local python env, might have time to fix and test later today otherwise later this week. For the time being I pushed a commit that changes to Hive style where dropping and creating external tables in [helpers_iceberg](https://github.com/dbt-athena/dbt-athena/blob/main/dbt/include/athena/macros/materializations/models/helpers_iceberg.sql) which I know are required. If you have the time and feel for it you can give it a try again otherwise I'll look into it later.

**Comment by cstruct** [comment]
For those that are interested the reason this broke support for iceberg tables is that previously `relation.quote_character` was set to an empty string causing no quoting. This PR updates it to the Presto quote char `"` causing all identifiers to be quoted with it unless you explicitly call `relation.render_hive()` instead of implicit `relation.__str__()` or explicit `relation.render()`.

**Comment by nicor88** [comment]
@cstruct any plan to move this forward?

**Comment by cstruct** [comment]
Yes, I do plan to revisit this PR when I have the time and energy to do so, I can't give any guarantees as to when that will happen though. If you feel that you do not want to wait for that, the `Allow edits and access to secrets by maintainers` is checked so any maintainer can pick up form where I left off.

**Comment by BrechtDeVlieger** [comment]
What is still to be done for this PR to be merged? Are the latest changes to [helpers_iceberg](https://github.com/dbt-athena/dbt-athena/blob/main/dbt/include/athena/macros/materializations/models/helpers_iceberg.sql) tested @nicor88? I'm willing to continue the work on this if need be. Just not sure if the iceberg tests are still failing or not..

**Comment by nicor88** [comment]
@BrechtDeVlieger `helper_iceberg` doesn't exist anymore, this PR require a rebase with main and a fix for supporting iceberg.

**Pull request closed without merging**